src/controllers/food.controller.js

```diff
@@ -0,0 +1,343 @@
+import { ApiResponse } from "../utils/apiResponse.js";
+import { asyncHandler } from "../utils/asyncHandler.js";
+import {
+  getFoodItemById,
+  createFoodItem,
+  getFoodItemByName,
+} from "../db/foodItem.query.js";
+import {
+  createPrepares,
+  getPreparesById,
+  updatePreparesDiscountById,
+  deletePreparesById,
+} from "../db/prepares.query.js";
+import { getRestaurantById } from "../db/restaurant.query.js";
+
+const getFood = asyncHandler(async (req, res) => {
+  const { foodItemId, restaurantId } = req.body;
+
+  if (!foodItemId || !restaurantId) {
+    return res.status(400).json(
+      new ApiResponse(
+        400,
+        {
+          reason: foodItemId
+            ? "restaurantId not found"
+            : "foodItemId not found",
+          at: "foodItem.controller.js -> getFoodItem",
+        },
+        "Bad request."
+      )
+    );
+  }
+
+  try {
+    const [foodItem, preparesItem, restaurant] = await Promise.all([
+      getFoodItemById(foodItemId),
+      getPreparesById(foodItemId, restaurantId),
+      getRestaurantById(restaurantId),
+    ]);
+
+    const data = {
+      foodItem: foodItem[0],
+      preparesItem: preparesItem[0],
+      restaurant: restaurant[0],
+    };
+
+    for (const key in data) {
+      if (!data[key]) {
+        return res.status(404).json(
+          new ApiResponse(
+            404,
+            {
+              reason: `${key} could not be found using the provided ID.`,
+              at: "foodItem.controller.js -> getFood",
+            },
+            "Item not found."
+          )
+        );
+      }
+    }
+
+    return res
+      .status(200)
+      .json(new ApiResponse(200, data, "Food item retrieved successfully."));
+  } catch (error) {
+    return res.status(500).json(
+      new ApiResponse(
+        500,
+        {
+          reason: error.message || "Food item cannot be retrieved",
+          at: "foodItem.controller.js -> getFoodItem",
+        },
+        "Unable to retrieve item."
+      )
+    );
+  }
+});
+
+const addFood = asyncHandler(async (req, res) => {
+  const { restaurantId, name, type, price, foodImageUrl, description } =
```

Comment:
Is it possible to fetch `restaurantId` using `verifyJwt`?

Reply:
Will have to create a new middleware because it is not a user